src/code42cli/cmds/profile.py

```diff
@@ -57,32 +61,42 @@ def show(profile_name):
 @name_option
 @server_option
 @username_option
+@password_option
 @disable_ssl_option
-def create(name, server, username, disable_ssl_errors=False):
+def create(name, server, username, password, disable_ssl_errors):
     """Create profile settings. The first profile created will be the default."""
     cliprofile.create_profile(name, server, username, disable_ssl_errors)
-    _prompt_for_allow_password_set(name)
+    if password:
+        _set_pw(name, password)
+    else:
+        _prompt_for_allow_password_set(name)
     echo("Successfully created profile '{}'.".format(name))
 
 
 @profile.command()
 @name_option
 @server_option
 @username_option
+@password_option
 @disable_ssl_option
-def update(name=None, server=None, username=None, disable_ssl_errors=None):
+def update(name, server, username, password, disable_ssl_errors):
     """Update an existing profile."""
     profile = cliprofile.get_profile(name)
     cliprofile.update_profile(profile.name, server, username, disable_ssl_errors)
-    _prompt_for_allow_password_set(profile.name)
+    if password:
+        _set_pw(name, password)
+    else:
+        _prompt_for_allow_password_set(profile.name)
     echo("Profile '{}' has been updated.".format(profile.name))
 
 
 @profile.command()
 @profile_name_arg
-def reset_pw(profile_name=None):
+def reset_pw(profile_name):
```

Comment:
This is not directly related to your change, but I think we should clarify that this command does not actually reset your code42 password and instead simply changes the password that is stored with the profile.